### ISSUE TITLE
Display offline contacts with unread messages at the top

### ISSUE THREAD

**Issue by sudden6** [qTox/qTox#5595]
##### Brief Description

Commit hash:  https://github.com/qTox/qTox/commit/993f6fa5c0f97adf22af3f30a26f4e84188baefa

…

Reproducible: Always

##### Steps to reproduce

1.  friend sends a message
2.  I do not read the message
3.  friend goes offline 

##### Observed Behavior

The friend is displayed as offline friend and sorted alphabetically with the rest of the offline contacts, possibly at the very bottom of the list. I don't see I got a message because I would have to scroll.

##### Expected Behavior

The contact list would be sorted as follows:

1) Online friends with unread messages
2) Online friends
3) Offline friends with unread messages
4) Offline friends

or

1) Online friends with unread messages
2) Offline friends with unread messages
3) Online friends
4) Offline friends

**Comment by anthonybilinski** [comment]
I agree this is an issue and have hit it myself a few times. I think this is roughly related to https://github.com/qTox/qTox/issues/5035 since "activity" sorting mode seems like it could be really useful and take multiple things into account like unread message + online status + frequency of chat etc. and maybe even be the default sort mode.

I'm hesitant about changing the sorting of "By Name" to consider too many factors, since it sounds like it should be pretty purely alphabetical. One option to address this that wouldn't involve changing the alphabetical order any more would be to use the "Unread Messages" notification 
![image](https://user-images.githubusercontent.com/10469203/55274934-9da8ba80-529b-11e9-8ff8-7c3755780543.png)
more aggressively, and have it present anytime there's an unread notification off the scrollbar, including offline friends, instead of hiding it once the friend is seen in the list once.
